### PR TITLE
Issue 5781: specific option for profiling instead of -v abuse

### LINKED ISSUE #5781: New flag `--profile` replaces verbosity option `-vprofile`
I noticed that `-vprofile:7` overrides `-vprofile.definitions:10` and `-vprofile.modules:10`:

https://github.com/agda/agda/blob/ff82d4ab9009d5af8bd267661b65614564f63ed6/src/full/Agda/TypeChecking/Monad/Benchmark.hs#L60-L64

Is this intentional? It is not [documented](https://agda.readthedocs.io/en/v2.6.2.1/tools/performance.html#measuring-typechecking-performance).

### PR DESCRIPTION
**Pull request by UlfNorell**
fixes #5781

### CODE REVIEW

**Comment by gallais** [comment]
> +262 −70,142

Hmmmm

File: src/full/Agda/TypeChecking/Monad/Benchmark.hs [hunk 1]
```diff
@@ -22,26 +22,9 @@ import qualified Agda.Utils.Benchmark as B
...
-benchmarkModulesLevel = 10
-
-benchmarkDefsKey :: String
-benchmarkDefsKey = "profile.definitions"
-
-benchmarkDefsLevel :: Int
-benchmarkDefsLevel = 10
-
--- | When verbosity is set or changes, we need to turn benchmarking on or off.
+-- | When profile options are set or changes, we need to turn benchmarking on or off.
```
Comment (andreasabel) [inline src/full/Agda/TypeChecking/Monad/Benchmark.hs]: "changed"

File: src/full/Agda/TypeChecking/Monad/Benchmark.hs [hunk 2]
```diff
@@ -51,20 +34,14 @@ updateBenchmarkingStatus =
...
-    _            -> B.BenchmarkOff
+benchmarking = liftTCM $
+  ifM (hasProfileOption Profile.Internal)    (pure $ B.BenchmarkSome isInternalAccount) $
+  ifM (hasProfileOption Profile.Definitions) (pure $ B.BenchmarkSome isDefAccount) $
+  ifM (hasProfileOption Profile.Modules)     (pure $ B.BenchmarkSome isModuleAccount) $
+  pure B.BenchmarkOff
 
 -- | Prints the accumulated benchmark results. Does nothing if
--- profiling is not activated at level 2.
+--   no benchmark profiling is enabled.
```
Comment (andreasabel) [inline src/full/Agda/TypeChecking/Monad/Benchmark.hs:44]: Is the comment in sync with the code? (see below)

File: src/full/Agda/TypeChecking/Monad/Benchmark.hs [hunk 3]
```diff
@@ -76,9 +53,8 @@ print = liftTCM $ whenM (B.isBenchmarkOn [] <$> benchmarking) $ do
   -- Ulf, 2020-03-04: Using benchmarkLevel here means that it only prints if internal benchmarking
   -- is turned on, effectively making module/definition benchmarking impossible (since internal
   -- takes precedence). It needs to be > 1 to avoid triggering #2602 though. Also use
-  -- displayDebugMessage instead of reportSLn to avoid requiring -v profile:2 in addition to the
-  -- specific profile levels.
-  displayDebugMessage benchmarkKey 2 $ prettyShow b
+  -- displayDebugMessage instead of reportSLn to avoid requiring -v profile:2.
+  displayDebugMessage "profile" 2 $ prettyShow b
```
Comment (andreasabel) [inline src/full/Agda/TypeChecking/Monad/Benchmark.hs:57]: In sync with this? (see above)

Comment (UlfNorell) [reply src/full/Agda/TypeChecking/Monad/Benchmark.hs:57]: As far as I understand things, yes. I believe `displayDebugMessage` doesn't check the verbosity settings.

**Comment by nad** [comment]
The [old documentation](https://agda.readthedocs.io/en/v2.6.2.1/tools/performance.html#measuring-typechecking-performance) should also be updated.

File: src/full/Agda/TypeChecking/Monad/Benchmark.hs (hunk 3 above)
Comment (andreasabel) [reply src/full/Agda/TypeChecking/Monad/Benchmark.hs:57]: Ah, I see.  The verbosity level is passed to the interaction backend, but not checked.

**Pull request merged**